### PR TITLE
VDS-2877: Implement companies listing

### PR DESCRIPTION
**Pull request by ilyawzrd**
Description: 

--

QA-test: VDS-2231

Demo-test: VDS-2232

Download artifact URL: ghcr.io/virtocommerce/demo-xapi-app:1.0.0-pr-71-vds-2877-6c3ead6a

### CODE REVIEW

File: src/app/modules/companies/companies.component.html [hunk 1]
```diff
@@ -1 +1,45 @@
...
+            <th scope="col">Phone</th>
+            <th scope="col">Email</th>
+            <th scope="col">Status</th>
+            <th scope="col">Actions</th>
+          </tr>
+        </thead>
+        <tbody>
+          <tr *ngFor="let company of companies.items">
+            <td>{{company.name}}</td>
+            <td>{{company.address?.countryCode}},{{company.address?.regionName}},{{company.address?.city}}, {{company.address?.line1}}, {{company.address?.postalCode}}</td>
```
Comment (trueboroda) [inline src/app/modules/companies/companies.component.html]: Move to the controller method

File: src/app/modules/companies/components/companies-list-filter-bar/companies-list-filter-bar.component.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+})
+export class CompaniesListFilterBarComponent {
+  searchFilterValue = '';
+
+  selectInputValue = '';
+
+  @Output() filterChange = new EventEmitter<FilterValues>();
+
+  onValueChanges({ value, emitterElement } : InputValues): void {
+    switch (emitterElement) {
```
Comment (trueboroda) [inline src/app/modules/companies/components/companies-list-filter-bar/companies-list-filter-bar.component.ts]: Using a different events handler is more preferable.

File: src/app/modules/companies/components/search-filter/search-filter.component.html [hunk 3]
```diff
@@ -0,0 +1,6 @@
+<div class="input-group input-group-lg">
+  <input type="text" class="form-control" placeholder="Search" (input)="onSearchFilterUpdate($event)" />
```
Comment (trueboroda) [inline src/app/modules/companies/components/search-filter/search-filter.component.html:2]: Usage of angular's events is more preferable.

File: src/app/modules/companies/components/option-filter/option-filter.component.html [hunk 4]
```diff
@@ -0,0 +1,6 @@
+<div class="input-group input-group-lg">
+  <select class="form-select lh-base" id="optionFilter" (change)="onSelectInputUpdate($event)">
```
Comment (trueboroda) [inline src/app/modules/companies/components/option-filter/option-filter.component.html:2]: Usage of angular's events is more preferable.

File: src/app/modules/companies/constants/pageInfo.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+export const pageInfo = {
```
Comment (trueboroda) [inline src/app/modules/companies/constants/pageInfo.ts:1]: Maybe better move this consts to a common level.

**Review by trueboroda — APPROVED** [review]

File: src/app/modules/companies/companies.component.ts [hunk 6]
```diff
@@ -7,5 +14,75 @@ import { Component } from '@angular/core';
     './companies.component.scss',
   ],
 })
-export class CompaniesComponent {
+export class CompaniesComponent implements OnInit, OnDestroy {
+  unsubscriber = new Subject();
```
Comment (asvishnyakov) [inline src/app/modules/companies/companies.component.ts]: There is no subscriptions

**Review by asvishnyakov — CHANGES_REQUESTED** [review]

File: src/app/modules/companies/companies.component.ts (hunk 6 above)
Comment (ilyawzrd) [reply src/app/modules/companies/companies.component.ts]: done

File: src/app/modules/companies/constants/pageInfo.ts (hunk 5 above)
Comment (ilyawzrd) [reply src/app/modules/companies/constants/pageInfo.ts:1]: Currently we have separate settings for each module. We have a user story in backlog to control such things on a higher level

File: src/app/modules/companies/components/companies-list-filter-bar/companies-list-filter-bar.component.ts (hunk 2 above)
Comment (ilyawzrd) [reply src/app/modules/companies/components/companies-list-filter-bar/companies-list-filter-bar.component.ts]: done

File: src/app/modules/companies/companies.component.html (hunk 1 above)
Comment (ilyawzrd) [reply src/app/modules/companies/companies.component.html]: done

**Review by asvishnyakov — APPROVED** [review]

**Pull request merged**